### PR TITLE
Add the possibility to set a strain

### PR DESCRIPTION
**Pull request by DiegoDeGusem**
This code allows user to set a normal and/or shear strain. However, it is impossible to combine this with elastodynamics, because elastodynamics wants to modify strain based on displacement. If someone would try this, they get an error.
This pull request also includes a test where the magnetoelastic field from mumax⁺ is compared with mumax³ (the results are exactly the same up to floating points :sunglasses: ).

### CODE REVIEW

File: test/mumax3/test_mumax3_rigid_strain.py
Comment (ilateur) [inline test/mumax3/test_mumax3_rigid_strain.py:1]: nice, it might be a bit faster to reuse the mumax+ world and magnet across different tests though.

File: test/mumax3/test_mumax3_rigid_strain.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            exx = {exx}
+            eyy = {eyy}
+            ezz = {ezz}
+            exy = {exy}
+            exz = {exz}
+            eyz = {eyz}
+            B1 = {B}
+            B2 = {B}
+
+            m = Uniform{tuple(magnetization)}
```
Comment (ilateur) [inline test/mumax3/test_mumax3_rigid_strain.py:39]: Is this arbitrary enough? Maybe safer to use random?

File: test/mumax3/test_mumax3_rigid_strain.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        err = max_relative_error(result, wanted)
+        assert err < RTOL
+
+    def test_eyz(self):
+        exx, eyy, ezz, exy, exz, eyz = 0, 0, 0, 0, 0, strain
+        magnet, mumax3sim = simulation(exx, eyy, ezz, exy, exz, eyz)
+        wanted = mumax3sim.get_field("B")
+        result = magnet.magnetoelastic_field.eval()
+        err = max_relative_error(result, wanted)
+        assert err < RTOL
```
Comment (ilateur) [inline test/mumax3/test_mumax3_rigid_strain.py:110]: Maybe also a test with combined (and negative?) strains?

File: src/physics/magnetoelasticfield.hpp [hunk 3]
```diff
@@ -8,6 +8,7 @@ class Field;
 
 // Assure that the magnetoelastic field and force are 0
 bool magnetoelasticAssuredZero(const Ferromagnet*);
+bool appliedStrain(const Magnet*);
```
Comment (ilateur) [inline src/physics/magnetoelasticfield.hpp:11]: I would call it `rigidStrainAssuredZero` (and return the opposite), in line with all other assuredZero functions.

**Review by ilateur — CHANGES_REQUESTED** [review]

File: mumaxplus/magnet.py [hunk 4]
```diff
@@ -303,6 +308,52 @@ def rho(self):
...
+        See Also
+        --------
+        elastic_energy, elastic_energy_density, elastic_displacement, stress_tensor
+        """
+        return Parameter(self._impl.rigid_norm_strain)
+    
+    @rigid_norm_strain.setter
+    def rigid_norm_strain(self, value):
+        if self.enable_elastodynamics:
+            raise Exception("You can not use elastodynamics with normal strain.")
```
[the remark this answers is not in the document]
Comment (DiegoDeGusem) [reply mumaxplus/magnet.py:331]: What do you mean?

[the remark this answers is not in the document]
Comment (ilateur) [reply mumaxplus/magnet.py:331]: "You can not use normal strain with elastodynamics", because it's a normal strain error

File: test/mumax3/test_mumax3_rigid_strain.py
Comment (ilateur) [reply test/mumax3/test_mumax3_rigid_strain.py:1]: See `test_parameter` for example for a nice useage of `@pytest.fixture`

File: src/physics/magnetoelasticfield.cu [hunk 5]
```diff
@@ -51,26 +55,65 @@ __global__ void k_magnetoelasticField(CuField hField,
   }
 }
 
+__global__ void k_rigidMagnetoelasticField(CuField hField,
+                                           const CuField mField,
+                                           const CuField normStrain,
+                                           const CuField shearStrain,
```
Comment (ilateur) [inline src/physics/magnetoelasticfield.cu:61]: They should be CuVectorParameters

**Review by ilateur — APPROVED** [review]
